### PR TITLE
i18n(pt-BR): Add deploy/github.md translation

### PR DESCRIPTION
**Pull request by ralacerda**
- Translated content

Description

Add pt-BR [deploy/github.md](https://docs.astro.build/en/guides/deploy/github/) translation.

I did not translate "deploy", "build", "commit" and "push", following suggestions on #1679. 

I also did not translate "Actions" since it is used in the context of Github Actions. 

I was unsure about translating instructions for the Github website. I couldn't found an option to change language on Github, so I'm not even sure if there is an official translation for the website. Please let me know if there is one so I can use it as a reference.

And I'm also participating on the Hacktoberfest. :smile:

### CODE REVIEW

File: src/pages/pt-br/guides/deploy/github.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+   A [action](https://github.com/withastro/action) oficial do Astro escaneia por lockfile para detectar o seu gerenciador de pacotes favorito (`npm`, `yarn`, or `pnpm`). Você deve realizar o commit do `package-lock.json`, `yarn.lock`, ou `pnpm-lock.yaml` criado automaticamente pelo seu pacote favorito, no seu repositório.
+   :::
+
+3. No GitHub,vá para a aba de **Settings** e encontre a sessão de **Pages**.
+
+4. Escolha **GitHub Actions** como o **Source** do seu site e aperte **Save**.
+
+5. Faça o commit de um novo arquivo de workflow e faça um push para o Github.
+
+O seu site já deve estar publicado! Quando você fizer um push com suas mudanças para o seu repositório do projeto Astro, as Github Actions vão automaticamente fazer o deploy para você.
```
Comment (Yan-Thomas) [inline src/pages/pt-br/guides/deploy/github.md:95]: Algumas sugestões de mudança na escrita para melhorar o entendimento
```suggestion
O seu site deve estar publicado agora! Quando você fizer push de mudanças para o repositório do seu projeto Astro, a Action do GitHub irá fazer deploy delas automaticamente para você.
```

File: src/pages/pt-br/guides/deploy/github.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+:::tip[determinar um domínio personalizado]
+Você pode opcionalmente determinar um domínio personalizado adicionando, no arquivo `./public/CNAME` do seu projeto, o seguinte:
```
Comment (Yan-Thomas) [inline src/pages/pt-br/guides/deploy/github.md:98]:
```suggestion
:::tip[Configure um domínio personalizado]
Você pode opcionalmente configurar um domínio personalizado adicionando o arquivo `./public/CNAME` a seguir no seu projeto:
```

File: src/pages/pt-br/guides/deploy/github.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+```txt title="public/CNAME"
+meu.subdominio.com
+```
+
+Isso vai realizar o deploy do seu site no seu domínio personalizado ao invés de `usuario.github.io`. Não se esqueça de [configurar o DNS no seu provedor de domínios](https://docs.github.com/pt/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).
```
Comment (Yan-Thomas) [inline src/pages/pt-br/guides/deploy/github.md:104]:
```suggestion
Isso vai realizar o deploy do seu site no seu domínio personalizado ao invés de `usuario.github.io`. Não se esqueça de também [configurar o DNS no seu provedor de domínios](https://docs.github.com/pt/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).
```

**Review by Yan-Thomas — COMMENTED** [review]
Thanks @ralacerda, great job! I added a couple suggestions for you. 🙌

**Comment by ralacerda** [comment]
Thanks for the suggestions @Yan-Thomas, I think they all made sense and the comments will help me in future translations.
I've commited the suggestions and visually checked the preview for regressions. 

If everything is okay I would appreciate if the `hacktoberfest-accepted` label was set. :smile:

**Pull request merged**